base/array.jl

```diff
@@ -13,7 +25,18 @@ const IntOrInd = Union{Int, AbstractUnitRange}
 const DimsOrInds{N} = NTuple{N,DimOrInd}
 const NeedsShaping = Union{Tuple{Integer,Vararg{Integer}}, Tuple{OneTo,Vararg{OneTo}}}
 
+"""
+    Vector{T}
+
+Vector type containing elements of type `T`. Alias for [`Array{T,1}`](@ref).
+"""
 const Vector{T} = Array{T,1}
+
+"""
+    Matrix{T}
+
+Matrix type containing elements of type `T`. Alias for [`Array{T,2}`](@ref).
```

Comment:
These descriptions of `Vector`, `Matrix`, `AbstractVector`, and `AbstractMatrix` seem a bit tautological ("[a `Vector` is a] Vector type containing elements of type `T`."). Perhaps something the lines of (for `Vector`) "[a `Vector` is a] one-dimensional dense array with elements of type `T`, often used to represent a mathematical vector." would be slightly more informative?